src/isar/apis/models/start_mission_definition.py

```diff
@@ -57,17 +60,31 @@ def to_isar_mission(mission_definition: StartMissionDefinition) -> Mission:
                     tag_id=tag_id,
                     analysis=inspection.analysis_types,
                     metadata=inspection.metadata,
+                    id=inspection.id,
                 )
                 for inspection in task.inspections
             ]
         except ValueError as e:
             raise MissionPlannerError(f"Failed to create task: {str(e)}")
+
+        if contains_duplicate_ids(items=inspection_steps):
+            raise MissionPlannerError(
+                f"Failed to create task: Duplicate step IDs are not allowed"
+            )
```

Comment:
Do we allow duplicate ideas in steps from different tasks?

Reply:
Yes, currently, but that was not intended, good catch.
Will fix

Reply:
Fixed